### PR TITLE
[STORM-1232] port backtype.storm.scheduler.DefaultScheduler to java

### PR DESCRIPTION
**Pull request by vesense**

### CODE REVIEW

**Comment by vesense** [comment]
Now the `DefaultScheduler` code is dependent on java `EvenScheduler`. This may cause compilation errors. I'll fix it later.

File: storm-core/src/jvm/org/apache/storm/scheduler/DefaultScheduler.java [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+        if (numWorkers != 0) {
+            Map<Integer, Integer> distribution = new TreeMap<Integer, Integer>();
+            int quotient = numExecutors / numWorkers;
+            int remainder = numExecutors % numWorkers;
+            distribution.put(quotient, numWorkers - remainder);
+            if(remainder != 0) {
+                distribution.put(quotient + 1, remainder);
+            }
+
+            Map<WorkerSlot, List<ExecutorDetails>> _slots = new HashMap<WorkerSlot, List<ExecutorDetails>>();
```
Comment (abhishekagarwal87) [inline storm-core/src/jvm/org/apache/storm/scheduler/DefaultScheduler.java]: map is not needed, set<WorkerSlot> would suffice.

File: storm-core/src/jvm/org/apache/storm/scheduler/DefaultScheduler.java [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+import java.util.Map;
+import java.util.Map.Entry;
+import java.util.Set;
+import java.util.TreeMap;
+
+public class DefaultScheduler implements IScheduler {
+
+    private Set<WorkerSlot> badSlots(Map<WorkerSlot, List<ExecutorDetails>> existingSlots, int numExecutors, int numWorkers) {
+        if (numWorkers != 0) {
+            Map<Integer, Integer> distribution = new TreeMap<Integer, Integer>();
```
Comment (abhishekagarwal87) [inline storm-core/src/jvm/org/apache/storm/scheduler/DefaultScheduler.java]: this logic is in Utils.java (integerDivided) do you want to reuse that?

Comment (vesense) [inline storm-core/src/jvm/org/apache/storm/scheduler/DefaultScheduler.java]: Good catch. Updated.

File: storm-core/src/jvm/org/apache/storm/scheduler/DefaultScheduler.java (hunk 1 above)
Comment (vesense) [inline storm-core/src/jvm/org/apache/storm/scheduler/DefaultScheduler.java]: Yeah. Set should suffice. Fixed.

**Comment by vesense** [comment]
Thanks @abhishekagarwal87 for your review. Comments addressed.

**Comment by vesense** [comment]
Since resolving conflicts merge cause some issues, I have created a new PR(https://github.com/apache/storm/pull/1108). Please take a look.

**Comment by vesense** [comment]
The new PR also fixed the dependence issue.

**Pull request closed without merging**